### PR TITLE
feat: cli basic framework

### PR DESCRIPTION
**Pull request by 0xff-dev**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it
feat: cli basic framework

#### Which issue(s) this PR fixes

For #315

### CODE REVIEW

File: pkg/cmd/create_cluster.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+)
+
+func NewCreateClusterCmd() *cobra.Command {
```
Comment (bjwswang) [inline pkg/cmd/create_cluster.go:25]: For the 1st version,should we manage cluster ?

Comment (0xff-dev) [reply pkg/cmd/create_cluster.go:25]: If we need cli to manage clusters, I think we need to add handling of clusters and can just introduce kind as our subcommand.

**Review by Abirdcfly — APPROVED** [review]

File: cmd/core/main.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+var longUsage = `Quickly deploy the kubebb platform in the existing k8s environment.
+
+The platform mainly includes four components: core, cluster-component, u4a, and component-store.
+
+For more details, please refer to https://kubebb.github.io/website
+
+Examples:
+    # create a cluster
+    corectl install --cluster-comopnent=true --component-store=true
```
Comment (bjwswang) [inline cmd/core/main.go:33]: How about  `corectl install --context ~/.kube/devconfig` with all four components installed by defualt.

File: cmd/core/main.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+)
+
+var longUsage = `Quickly deploy the kubebb platform in the existing k8s environment.
+
+The platform mainly includes four components: core, cluster-component, u4a, and component-store.
+
+For more details, please refer to https://kubebb.github.io/website
+
+Examples:
+    # create a cluster
```
Comment (bjwswang) [inline cmd/core/main.go:32]: install kubebb to a cluster

File: pkg/cmd/config.go [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+	var (
+		installClusterComponent, installU4A, installComponentStore bool
+	)
+	cmd := &cobra.Command{
+		Use:  "install",
+		Long: "install core, cluster-component",
+		Run: func(cmd *cobra.Command, args []string) {
+			initConf := config{
+				// Must install core
+				installers: map[string][]string{
```
Comment (bjwswang) [inline pkg/cmd/config.go:72]: how about place all components into installers by default?

**Review by Abirdcfly — APPROVED** [review]

**Review by bjwswang — APPROVED** [review]

**Comment by bjwswang** [comment]
@0xff-dev we can continue on this task based on this pr https://github.com/kubebb/core/pull/353

**Pull request merged**